template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hue/HueConfigProvider.java

```diff
@@ -72,4 +81,30 @@ protected DatabaseType dbType() {
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         return List.of();
     }
+
+    private void configureKnoxProxyHostsServiceConfig(TemplatePreparationObject source, List<ApiClusterTemplateConfig> result) {
+        GatewayView gateway = source.getGatewayView();
+        GeneralClusterConfigs generalClusterConfigs = source.getGeneralClusterConfigs();
```

Comment:
you could push down these to the .externalFQDNShouldConfigured()
otherwise LGTM